### PR TITLE
Update spacer.dart

### LINKED ISSUE #98787: [Proposal] Spacer Widget should extends from Flexible
Now in Flutter framework ，Spacer is extends StatelessWidget , and use Expanded Widget when build. Is unnecessary. just like Expanded Widget, extends Flexible is fine.   

now, `spacer.dart` file in https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/widgets/spacer.dart

image

### PR DESCRIPTION
**Pull request by toly1994328**
Fixes: [flutter#98787](https://github.com/flutter/flutter/issues/98787)

### CODE REVIEW

File: packages/flutter/lib/src/widgets/spacer.dart [hunk 1]
```diff
@@ -38,29 +38,15 @@ import 'framework.dart';
...
-
-  @override
-  Widget build(BuildContext context) {
-    return Expanded(
-      flex: flex,
-      child: const SizedBox.shrink(),
-    );
-  }
+class Spacer extends Flexible {
+  const Spacer({
```
Comment (AlexV525) [inline packages/flutter/lib/src/widgets/spacer.dart:42]: Missing comments

```suggestion
  /// Creates a flexible space to insert into a [Flexible] widget.
  ///
  /// The [flex] parameter may not be null or less than one.
  const Spacer({
```

File: packages/flutter/lib/src/widgets/spacer.dart [hunk 2]
```diff
@@ -38,29 +38,15 @@ import 'framework.dart';
...
+    Key? key,
+    int flex = 1,
+  })  : assert(flex > 0),
+        super(
+          key: key,
+          flex: flex,
+          fit: FlexFit.tight,
+          child: const SizedBox.shrink(),
+        );
```
Comment (AlexV525) [inline packages/flutter/lib/src/widgets/spacer.dart:51]:
```suggestion
  }) : assert(flex > 0),
       super(
         key: key,
         flex: flex,
         fit: FlexFit.tight,
         child: const SizedBox.shrink(),
       );
```

**Review by AlexV525 — COMMENTED** [review]
Some format tips.

**Comment by goderbauer** [comment]
What's the rational for this change? I think, overall Flutter prefers composition over inheritance.

It appears that there's also a duplicated PR for this: https://github.com/flutter/flutter/pull/98921

**Comment by Hixie** [comment]
Closing for reasons given in the issue.

**Pull request closed without merging**